verifier/src/channel.rs

```diff
@@ -237,18 +281,25 @@ where
 
 /// Container of trace query data, including:
 /// * Queried states for all trace segments.
-/// * Merkle authentication paths for all queries.
+/// * Batch opening proof for all queries.
 ///
 /// Trace states for all auxiliary segments are stored in a single table.
-struct TraceQueries<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> {
-    query_proofs: Vec<BatchMerkleProof<H>>,
+struct TraceQueries<
+    E: FieldElement,
+    H: ElementHasher<BaseField = E::BaseField>,
+    V: VectorCommitment<H>,
+> {
+    query_proofs: Vec<V::MultiProof>,
     main_states: Table<E::BaseField>,
     aux_states: Option<Table<E>>,
+    _h: PhantomData<H>,
 }
 
-impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>> TraceQueries<E, H> {
+impl<E: FieldElement, H: ElementHasher<BaseField = E::BaseField>, V: VectorCommitment<H>>
+    TraceQueries<E, H, V>
+{
```

Comment:
Not for this PR, but I think it would be cleaner to make all such instances consistent and look like so:
```rust
struct TraceQueries<E, H, V>
where
    E: FieldElement,
    H: ElementHasher<BaseField = E::BaseField>,
    V: VectorCommitment<H>,
{
    query_proofs: Vec<V::MultiProof>,
    main_states: Table<E::BaseField>,
    aux_states: Option<Table<E>>,
    _h: PhantomData<H>,
}

impl<E, H, V> TraceQueries<E, H, V>
where
    E: FieldElement,
    H: ElementHasher<BaseField = E::BaseField>,
    V: VectorCommitment<H>,
{
```

Reply:
Have changed it here and all other places I could find